src/spork.cpp

```diff
@@ -145,24 +189,27 @@ bool CSporkManager::IsSporkActive(int nSporkID)
     LOCK(cs);
     int64_t r = -1;
 
-    if(mapSporksActive.count(nSporkID)){
-        r = mapSporksActive[nSporkID].nValue;
-    } else if (mapSporkDefaults.count(nSporkID)) {
-        r = mapSporkDefaults[nSporkID];
-    } else {
-        LogPrint("spork", "CSporkManager::IsSporkActive -- Unknown Spork ID %d\n", nSporkID);
-        r = 4070908800ULL; // 2099-1-1 i.e. off by default
+    if(SporkValueIsActive(nSporkID, r)){
+        return r < GetAdjustedTime();
+    }
+
+    if (mapSporkDefaults.count(nSporkID)) {
+        return  mapSporkDefaults[nSporkID] < GetAdjustedTime();
     }
 
+    LogPrint("spork", "CSporkManager::IsSporkActive -- Unknown Spork ID %d\n", nSporkID);
+    r = 4070908800ULL; // 2099-1-1 i.e. off by default
     return r < GetAdjustedTime();
 }
 
 // grab the value of the spork on the network, or the default
 int64_t CSporkManager::GetSporkValue(int nSporkID)
 {
     LOCK(cs);
-    if (mapSporksActive.count(nSporkID))
-        return mapSporksActive[nSporkID].nValue;
+    int64_t r = -1;
```

Comment:
I'd prefer a better name

Reply:
@PastaPastaPasta , I'm writing wrong `if`s automatically,  I think it is bad practice to break codestyle guide,  so it is good you are pointing this errors to me, I'm sorry that I'm making them so often.)